idea/idea-frontend-fir/src/org/jetbrains/kotlin/idea/frontend/api/fir/components/KtFirSmartcastProvider.kt

```diff
@@ -50,4 +54,22 @@ internal class KtFirSmartcastProvider(
             }?.let(::add)
         }
     }
+
+    override fun getStableNullability(expression: KtExpression): StableNullability {
+        val fir = expression.getOrBuildFir(analysisSession.firResolveState)
+        // TODO: Check stability of smartcast (pending PR 4382)
```

Comment:
FYI, #4382 merged

Reply:
@punzki are you going to address this TODO in this PR?

Reply:
Ah, that one is reverted, so I don't think we can address this at the moment.

Reply:
Yes, I was about to do so until I found out it was reverted.